pyblazing/pyblazing/apiv2/context.py

```diff
@@ -1709,10 +1718,23 @@ def sql(self, query, table_list=[], algebra=None, return_futures=False, single_g
                             query_config_options,
                             workers=[worker]))
                     i = i + 1
+
                 if(return_futures):
                     result  = dask_futures
                 else:
+                    # Check for crashed workers
+                    while any([not f.done() for f in dask_futures]):
+
+                        workers_info = self.dask_client.scheduler_info()["workers"]
+                        last_seen_times = [workers_info[worker]['last_seen'] for worker in workers_info]
+                        if any(time > start_query_time for time in last_seen_times):
+                            raise RuntimeError("A worker crashed during query execution")
+                            break
+
+                        time.sleep(0.5)
```

Comment:
This sleep here will make it so that the fastest a query will return is 0.5 sec, followed by 1 sec, etc.... We should at the very least make this sleep much smaller like 0.1 or 0.2 sec, or figure out something other than a spin lock here